### PR TITLE
UX changes done for better view on mobile screen.

### PR DESCRIPTION
**Pull request by v-pankasahu**
**Objective:** 
Northwind Orders sample app should work across the application hub. 
Currently this is functionally working well. But data populating on screen are not properly aligned on mobile screen. Also, not in proper format. Objective for this issue need to be fixes with minimal changes so that other functionality should not be break and it will support well on web as well.

**Fixes done:**

- We have identified the screens and the element which were not properly aligned.
- We have added /overwrite the base CSS class (northwind-team.css) by adding custom style to achieve this goal.
- We have applied Id wise css on every distracted element on page.

**_Details Description along with sample video can be found on ADO task's attachment #_** [https://office.visualstudio.com/MetaOS/_workitems/edit/6336325/]

### CODE REVIEW

File: experimental/ExtendTeamsforM365/client/pages/categories.html [hunk 1]
```diff
@@ -6,6 +6,15 @@
     <title>Northwind Product Categories</title>
     <link rel="stylesheet" href="/northwind.css" />
     <link rel="icon" href="data:;base64,="> <!-- Suppress favicon error -->
+    <style>               
+        #categoriesImgae{
```
Comment (bhups-83) [inline experimental/ExtendTeamsforM365/client/pages/categories.html:10]: We already have categoryImage class, let's use that instead,

File: experimental/ExtendTeamsforM365/client/pages/categories.html [hunk 2]
```diff
@@ -6,6 +6,15 @@
     <title>Northwind Product Categories</title>
     <link rel="stylesheet" href="/northwind.css" />
     <link rel="icon" href="data:;base64,="> <!-- Suppress favicon error -->
+    <style>               
+        #categoriesImgae{        
+        width: 4rem;
+        }   
+
+        #categoriesDescription{
```
Comment (bhups-83) [inline experimental/ExtendTeamsforM365/client/pages/categories.html:14]: Let's not add inline css in the html, instead add it to CSS file. (northwind-teams.css)

File: experimental/ExtendTeamsforM365/client/pages/categories.html [hunk 3]
```diff
@@ -15,7 +24,7 @@
     <top-nav-panel></top-nav-panel>
 
     <h3>Northwind Product Categories</h3>
-    <table id="categories">
+    <table id="categories" style="overflow: auto;">
```
Comment (bhups-83) [inline experimental/ExtendTeamsforM365/client/pages/categories.html:27]: Move inline css to css file.

File: experimental/ExtendTeamsforM365/client/pages/categoryDetail.html [hunk 4]
```diff
@@ -6,6 +6,17 @@
     <title>Northwind Product Category Detail</title>
     <link rel="stylesheet" href="/northwind.css" />
     <link rel="icon" href="data:;base64,="> <!-- Suppress favicon error -->
+    <style>
```
Comment (bhups-83) [inline experimental/ExtendTeamsforM365/client/pages/categoryDetail.html:9]: Move this to css file.

File: experimental/ExtendTeamsforM365/client/pages/categoryDetail.html [hunk 5]
```diff
@@ -17,7 +28,7 @@
     <div id="content">
         <p>Loading...</p>
     </div>
-    <table id="products">
+    <table id="products" style="overflow: auto;">
```
Comment (bhups-83) [inline experimental/ExtendTeamsforM365/client/pages/categoryDetail.html:31]: Move this to css file.

File: experimental/ExtendTeamsforM365/client/pages/myOrders.html [hunk 6]
```diff
@@ -5,6 +5,29 @@
     <meta charset="UTF-8" />
     <title>Northwind Orders</title>   
     <link rel="icon" href="data:;base64,="> <!-- Suppress favicon error -->
+    <style>
```
Comment (bhups-83) [inline experimental/ExtendTeamsforM365/client/pages/myOrders.html:8]: move to css file.

File: experimental/ExtendTeamsforM365/client/pages/myOrders.html [hunk 7]
```diff
@@ -14,13 +37,13 @@
 <div class="container">
    
     <div class="element" id="allOrders" style="display: none;">
-        <div class="content" id="content">
+        <div class="content" id="content" >
```
Comment (bhups-83) [inline experimental/ExtendTeamsforM365/client/pages/myOrders.html:40]: drop leading whitespace. after content".

File: experimental/ExtendTeamsforM365/client/pages/myOrders.js [hunk 8]
```diff
@@ -40,12 +40,12 @@ const displayAllMyOrders=(employee)=> {
...
         orderRow.innerHTML = `<tr>           
             <td><a href="/pages/orderDetail.html?orderId=${order.orderId}">${order.orderId}</a></td>
-            <td>${(new Date(order.orderDate)).toDateString()}</td>
-            <td>${order.customerName}</td>
-            <td>${order.customerContact}</td>
-            <td>${order.customerPhone}</td>
+            <td id='ordersDate'>${(new Date(order.orderDate).toLocaleDateString('en-us', { weekday:"short", year:"numeric", month:"short", day:"numeric"}))}</td>
+            <td >${order.customerName}</td>
+            <td id='ordersCustomerName'>${order.customerContact}</td>
+            <td id='ordersPhoneNo'>${(order.customerPhone)}</td>
```
Comment (bhups-83) [inline experimental/ExtendTeamsforM365/client/pages/myOrders.js:46]: are parentheses required around orders.customerPhone? Other places as well.

File: experimental/ExtendTeamsforM365/client/pages/productDetail.html [hunk 9]
```diff
@@ -6,6 +6,24 @@
     <title>Northwind Product</title>
     <link rel="stylesheet" href="/northwind.css" />
     <link rel="icon" href="data:;base64,="> <!-- Suppress favicon error -->
+    <style>
```
Comment (bhups-83) [inline experimental/ExtendTeamsforM365/client/pages/productDetail.html:9]: Move to css file.

File: experimental/ExtendTeamsforM365/client/pages/productDetail.html [hunk 10]
```diff
@@ -17,9 +35,9 @@
     <div id="content">
         <p>Loading...</p>
     </div>
-    <table id="orders">
+    <table id="orders" style="overflow: auto;" style="overflow: auto;">
```
Comment (bhups-83) [inline experimental/ExtendTeamsforM365/client/pages/productDetail.html:38]: move to css file, other places as well.

File: experimental/ExtendTeamsforM365/client/styles/northwind-teams.css [hunk 11]
```diff
@@ -228,15 +228,18 @@ button#btnTaskModule {
...
+  width: 58%;
+}
+div#allOrders, div#rODiv {
+  width: 59%;
 }
 
 div#content {
   background: #ddd3;
-  padding: 10px;
+  padding: 10px;
```
Comment (bhups-83) [inline experimental/ExtendTeamsforM365/client/styles/northwind-teams.css:242]: drop trailing whitespace. after ;.

**Review by bhups-83 — CHANGES_REQUESTED** [review]

**Comment by rabwill** [comment]
Hey @v-pankasahu , I can see there is already some reviewer comments. I'll leave it to you folks to resolve them before checking further  :)

File: experimental/ExtendTeamsforM365/client/styles/northwind-teams.css (hunk 11 above)
Comment (rabwill) [reply experimental/ExtendTeamsforM365/client/styles/northwind-teams.css:242]: Should detect mobile devices and change width to 100% on mobile and devices for div#allOrders ? @bhups-83

File: experimental/ExtendTeamsforM365/client/pages/categories.js [hunk 12]
```diff
@@ -14,9 +14,9 @@ async function displayUI() {
             categories.forEach(category => {
                 const categoryRow = document.createElement('tr');
                 categoryRow.innerHTML = `<tr>
-                <td><img src="data:image/bmp;base64,${category.picture}" class="categoryImage"></image></td>
+                <td id='categoriesImgae'><img src="data:image/bmp;base64,${category.picture}" class="categoryImage"></image></td>
```
Comment (rabwill) [inline experimental/ExtendTeamsforM365/client/pages/categories.js:17]: Is this is a typo `categoriesImgae` in the id @v-pankasahu ?

File: experimental/ExtendTeamsforM365/client/pages/categories.html (hunk 1 above)
Comment (v-pankasahu) [reply experimental/ExtendTeamsforM365/client/pages/categories.html]: We changed the naming convention; it was used for ID purpose only.

File: experimental/ExtendTeamsforM365/client/pages/categories.html (hunk 2 above)
Comment (v-pankasahu) [reply experimental/ExtendTeamsforM365/client/pages/categories.html]: Changed and shifted to northwind-teams.css

File: experimental/ExtendTeamsforM365/client/pages/categories.html (hunk 3 above)
Comment (v-pankasahu) [reply experimental/ExtendTeamsforM365/client/pages/categories.html]: Changed and shifted to northwind-teams.css

File: experimental/ExtendTeamsforM365/client/pages/categoryDetail.html (hunk 4 above)
Comment (v-pankasahu) [reply experimental/ExtendTeamsforM365/client/pages/categoryDetail.html]: Changed and shifted to northwind-teams.css

File: experimental/ExtendTeamsforM365/client/pages/categoryDetail.html (hunk 5 above)
Comment (v-pankasahu) [reply experimental/ExtendTeamsforM365/client/pages/categoryDetail.html]: Changed and shifted to northwind-teams.css

File: experimental/ExtendTeamsforM365/client/pages/myOrders.html (hunk 6 above)
Comment (v-pankasahu) [reply experimental/ExtendTeamsforM365/client/pages/myOrders.html]: Changed and shifted to northwind-teams.css

File: experimental/ExtendTeamsforM365/client/pages/myOrders.html (hunk 7 above)
Comment (v-pankasahu) [reply experimental/ExtendTeamsforM365/client/pages/myOrders.html]: Dropped.

File: experimental/ExtendTeamsforM365/client/pages/productDetail.html (hunk 9 above)
Comment (v-pankasahu) [reply experimental/ExtendTeamsforM365/client/pages/productDetail.html]: Changed and shifted to northwind-teams.css

File: experimental/ExtendTeamsforM365/client/pages/productDetail.html (hunk 10 above)
Comment (v-pankasahu) [reply experimental/ExtendTeamsforM365/client/pages/productDetail.html]: Changed and shifted to northwind-teams.css

File: experimental/ExtendTeamsforM365/client/styles/northwind-teams.css (hunk 11 above)
Comment (v-pankasahu) [reply experimental/ExtendTeamsforM365/client/styles/northwind-teams.css]: Changes done for white space.

File: experimental/ExtendTeamsforM365/client/pages/categories.js (hunk 12 above)
Comment (v-pankasahu) [reply experimental/ExtendTeamsforM365/client/pages/categories.js]: Yeas, this was typo mistake. Corrected and renamed.

File: experimental/ExtendTeamsforM365/client/pages/myOrders.js (hunk 8 above)
Comment (v-pankasahu) [reply experimental/ExtendTeamsforM365/client/pages/myOrders.js]: removed.

**Comment by rabwill** [comment]
Looks like all comments resolved. Merging the changes. For any new updates pls open a new PR. @v-pankasahu Thanks for your work!

**Comment by Gso-1** [comment]
Hola yo los podría incorporar pero para ello necesito que se comuniquen conmigo a mi correo que es gasparsalinasoyaneder@educarparaavanzar.onmicrosoft.com

**Pull request merged**